core/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilter.java

```diff
@@ -15,33 +15,252 @@
 
+		RegisteredClient client = null;
+		OAuth2AuthorizationRequest authorizationRequest = null;
+		OAuth2Authorization authorization = null;
+
+		try {
+			checkUserAuthenticated();
+			client = fetchRegisteredClient(request);
+
+			authorizationRequest = authorizationRequestConverter.convert(request);
+			validateAuthorizationRequest(request, client);
+
+			String code = codeGenerator.generateKey();
+			authorization = buildOAuth2Authorization(client, authorizationRequest, code);
+			authorizationService.save(authorization);
+
+			String redirectUri = getRedirectUri(authorizationRequest, client);
+			sendCodeOnSuccess(request, response, authorizationRequest, redirectUri, code);
+		}catch(OAuth2AuthorizationException authorizationException) {
+			OAuth2Error authorizationError = authorizationException.getError();
+
+			if (authorizationError.getErrorCode().equals(OAuth2ErrorCodes.INVALID_REQUEST)
+					|| authorizationError.getErrorCode().equals(OAuth2ErrorCodes.ACCESS_DENIED)) {
+				sendErrorInResponse(response, authorizationError);
+			}
+			else if (authorizationError.getErrorCode().equals(OAuth2ErrorCodes.UNSUPPORTED_RESPONSE_TYPE)
+					|| authorizationError.getErrorCode().equals(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT)) {
+				String redirectUri = getRedirectUri(authorizationRequest, client);
+				sendErrorInRedirect(request, response, authorizationRequest, authorizationError, redirectUri);
+			}else {
+				throw new ServletException(authorizationException);
+			}
+		}
+
+	}
+
+	protected void checkUserAuthenticated() {
+		Authentication currentAuth = SecurityContextHolder.getContext().getAuthentication();
+		if (currentAuth==null || !currentAuth.isAuthenticated())
+			throw new OAuth2AuthorizationException(USER_NOT_AUTHENTICATED_ERROR);
+	}
+
+	protected RegisteredClient fetchRegisteredClient(HttpServletRequest request) throws OAuth2AuthorizationException {
+		String clientId = request.getParameter(OAuth2ParameterNames.CLIENT_ID);
+		if (StringUtils.isEmpty(clientId))
+			throw new OAuth2AuthorizationException(CLIENT_ID_ABSENT_ERROR);
+
+		RegisteredClient client = registeredClientRepository.findByClientId(clientId);
+		if (client==null)
+			throw new OAuth2AuthorizationException(CLIENT_ID_NOT_FOUND_ERROR);
+
+		boolean isAuthoirzationGrantAllowed = Stream.of(client.getAuthorizationGrantTypes())
+				.anyMatch(grantType -> grantType.contains(AuthorizationGrantType.AUTHORIZATION_CODE));
+			if (!isAuthoirzationGrantAllowed)
+				throw new OAuth2AuthorizationException(AUTHZ_CODE_NOT_SUPPORTED_ERROR);
+
+		return client;
+
+	}
+
+	protected OAuth2Authorization buildOAuth2Authorization(RegisteredClient client,
+			OAuth2AuthorizationRequest authorizationRequest, String code) {
+		OAuth2Authorization authorization = OAuth2Authorization.createBuilder()
+					.clientId(authorizationRequest.getClientId())
+					.addAttribute(OAuth2ParameterNames.CODE, code)
+					.addAttribute(OAuth2Authorization.ISSUED_AT, Instant.now())
```

Comment:
Please remove `ISSUED_AT`, `CODE_USED` and `SCOPE`. These will be added in a later story. We only need to store `CODE` and `OAuth2AuthorizationRequest` `attributes` as part of this story.

Reply:
Fixed in PR. Thanks.